tests/chainer_tests/test_variable.py

```diff
@@ -2742,26 +2742,47 @@ def test_grad_raise_double_backprop_2(self):
             chainer.grad([x.grad_var], [y.grad_var])
 
 
+@testing.backend.inject_backend_tests(None, _backend_params)
 class TestAsVariable(unittest.TestCase):
 
-    def check_to_variable_from_array(self, x):
+    def test_to_variable_from_array(self, backend_config):
+        x = backend_config.get_array(np.empty(1, np.float32))
         y = chainer.as_variable(x)
         assert isinstance(y, chainer.Variable)
-        assert y.data is x
-        assert not y.requires_grad
+        assert y.requires_grad is False
+
+        if backend_config.xp is chainerx:
+            # chainerx
+            assert y.array.shape == x.shape
+            assert y.array.device == x.device
+            assert y.array.strides == x.strides
+            assert not y.array.is_backprop_required()
+            chainerx.testing.assert_array_equal(y.array, x)
+        else:
+            # non-chainerx
+            assert y.array is x
 
-    def test_to_variable_from_numpy(self):
-        self.check_to_variable_from_array(np.empty(1, np.float32))
+    def test_to_variable_from_variable(self, backend_config):
+        x_arr = backend_config.get_array(np.empty(1, np.float32))
+        x = chainer.Variable(x_arr)
+        y = chainer.as_variable(x)
+        assert y is x
+        assert y.requires_grad is True
```

Comment:
This line is not required if the immutability of `requires_grad` is tested somewhere else.

The doc says "The requires_grad flag is kept".  How about parameterizing over `x.requires_grad`?
```
x = chainer.Variable(x_arr, requires_grad=requires_grad)
...
assert y.requires_grad is requires_grad
```

Reply:
Thanks, fixed.